gcp_variant_transforms/bq_to_vcf.py

```diff
@@ -164,30 +173,57 @@ def _bigquery_to_vcf_shards(
   `vcf_header_file_path`.
   """
   schema = _get_schema(known_args.input_table)
-  # TODO(allieychen): Modify the SQL query with the specified sample_ids.
-  query = _get_bigquery_query(known_args, schema)
+  query = _get_variant_query(known_args, schema)
   logging.info('Processing BigQuery query %s:', query)
-  bq_source = bigquery.BigQuerySource(query=query,
-                                      validate=True,
-                                      use_standard_sql=True)
+  project_id, dataset_id, table_id = bigquery_util.parse_table_reference(
+      known_args.input_table)
+  bq_variant_source = bigquery.BigQuerySource(query=query,
+                                              validate=True,
+                                              use_standard_sql=True)
   annotation_names = _extract_annotation_names(schema)
+
+  base_table_id = table_id[:table_id.find(TABLE_SUFFIX_SEPARATOR)]
+  sample_query = _SAMPLE_INFO_QUERY_TEMPLATE.format(PROJECT_ID=project_id,
+                                                    DATASET_ID=dataset_id,
+                                                    BASE_TABLE_ID=base_table_id)
+  bq_sample_source = bigquery.BigQuerySource(query=sample_query,
+                                             validate=True,
+                                             use_standard_sql=True)
   with beam.Pipeline(options=beam_pipeline_options) as p:
     variants = (p
-                | 'ReadFromBigQuery ' >> beam.io.Read(bq_source)
+                | 'ReadFromBigQuery ' >> beam.io.Read(bq_variant_source)
                 | bigquery_to_variant.BigQueryToVariant(annotation_names))
+    sample_table_rows = (
+        p
+        | 'ReadFromSampleTable' >> beam.io.Read(bq_sample_source))
     if known_args.sample_names:
-      sample_ids = (p
-                    | transforms.Create(known_args.sample_names,
-                                        reshuffle=False)
-                    | beam.combiners.ToList())
+      hash_table = (
+          sample_table_rows
+          | 'SampleNameToIdDict' >> sample_mapping_table.SampleNameToIdDict())
+      sample_names = (p
+                      | transforms.Create(known_args.sample_names,
+                                          reshuffle=False))
+      sample_ids = (sample_names
```

Comment:
To highlight the distinction between `sample_names` pcollection and list, we could perhaps rename this variable to `sample_ids_list` and the following variable also `sample_names_list`. Please feel free to come up with a better name than what I propose here, I myself don't like what I suggested :D

Reply:
hmm I don't know either... thought of consolidated_sample_ids but ended up with just combined_sample_ids. Did the same for _names. Tell me if not up to par.